src/Surfnet/StepupGateway/GatewayBundle/Service/StepUpAuthenticationService.php

```diff
@@ -139,33 +158,10 @@ public function determineViableSecondFactors(
             }
         }
 
-        $highestLoa = $this->resolveHighestLoa($loaCandidates);
-        if (!$highestLoa) {
-            $this->logger->info(sprintf(
-                'Out of %d candidate LoA\'s, no highest LoA could be determined; no candidate second factors',
-                count($loaCandidates)
-            ));
-            return new ArrayCollection();
+        if (!count($loaCandidates)) {
+            throw new RuntimeException('No loa can be found, at least one Loa (SP default) should be found');
         }
 
-        $this->logger->info(
-            sprintf('Out of %d candidate LoA\'s, LoA "%s" is the highest', count($loaCandidates), $highestLoa)
-        );
-
-        $candidateSecondFactors = $this->secondFactorRepository->getAllMatchingFor($highestLoa, $identityNameId);
-        $this->logger->info(
-            sprintf('Loaded %d matching candidate second factors', count($candidateSecondFactors))
-        );
-
-        return $candidateSecondFactors;
-    }
-
-    /**
-     * @param ArrayCollection $loaCandidates
-     * @return null|\Surfnet\StepupBundle\Value\Loa
-     */
-    private function resolveHighestLoa(ArrayCollection $loaCandidates)
-    {
         $actualLoas = new ArrayCollection();
         foreach ($loaCandidates as $loaDefinition) {
             $loa = $this->loaResolutionService->getLoa($loaDefinition);
```

Comment:
Doesn't this mean that when a SP requires a malformed LoA 'http://malformed.loa/loa5' this is ignored, and we fall back to a potentially lower LoA? Is that acceptable?